### PR TITLE
Clean some TODOs

### PR DESCRIPTION
**Pull request by zguesmi**
* Sort Voucher and VoucherHub functions
* Clean PoCo mock
* Rename hub storage variables
* Use same storage structure in Hub v2
* Add partial sponsoring test

### CODE REVIEW

File: test/beacon/Voucher.test.ts [hunk 1]
```diff
@@ -352,6 +352,49 @@ describe('Voucher', function () {
             expect(await voucherAsOwner.getSponsoredAmount(dealId)).to.be.equal(dealPrice);
         });
 
+        it('Should match orders with partial sponsored amount', async () => {
+            await addEligibleAssets([app, dataset]); // not workerpool.
```
Comment (jeremyjams) [inline test/beacon/Voucher.test.ts:356]: Should we merge classic and boost tests when possible?

Comment (zguesmi) [reply test/beacon/Voucher.test.ts:356]: Boost test was already implemented and there is a TODO to refactor boost and classic tests.

File: contracts/beacon/Voucher.sol [hunk 2]
```diff
@@ -472,4 +415,58 @@ contract Voucher is Initializable, IVoucher {
...
+        bytes32 dealId,
+        uint256 dealVolume,
+        address requester
+    ) private {
+        VoucherStorage storage $ = _getVoucherStorage();
+        require(!$._refundedTasks[taskId], "Voucher: task already refunded");
+        $._refundedTasks[taskId] = true;
+        if (taskPrice != 0) {
+            uint256 dealSponsoredAmount = $._sponsoredAmounts[dealId];
+            // A positive remainder is possible when the voucher balance is less than
```
[the remark this answers is not in the document]
Comment (zguesmi) [reply contracts/beacon/Voucher.sol:444]: Waiting for your merge to update this branch.

File: test/beacon/Voucher.test.ts [hunk 3]
```diff
@@ -352,6 +352,49 @@ describe('Voucher', function () {
...
+                .transfer(requester, dealSponsoredAmount)
+                .then((tx) => tx.wait());
+            // Allow voucher to spend non-sponsored amount.
+            await iexecPocoInstance
+                .connect(requester)
+                .approve(voucherAddress, dealPrice)
+                .then((tx) => tx.wait());
+            // Save initial balances.
+            const voucherInitialCreditBalance = await voucherAsOwner.getBalance();
+            const voucherInitialRlcBalance = await getVoucherBalanceOnIexecPoco();
```
[the remark this answers is not in the document]
Comment (zguesmi) [reply test/beacon/Voucher.test.ts]: Done.

File: test/beacon/Voucher.test.ts (hunk 1 above)
Comment (jeremyjams) [reply test/beacon/Voucher.test.ts:356]: Yes :)

**Pull request merged**